coalib/parsing/LineParser.py

```diff
@@ -55,6 +56,19 @@ def parse(self, line):
         line, comment = self.__separate_by_first_occurrence(
             line,
             self.comment_separators)
+
+        # Comment starting with # should have whitespace after #
+        # Example: C# is the hybrid of C and C++.
+        if '#' in line:
+            start_index = line.index('#')
+            if start_index == 0 and line[1] != ' ':
+                logging.warning('The comment line does not have \
+                                 whitespace after #')
+            elif line[start_index - 1] != ' ' \
+                    or line[start_index + 1] != ' ':
+                logging.warning('There is no whitespace either \
+                                 before or after # in comment')
+
```

Comment:
I am not an expert but perhaps a more descriptive warning can be given? perhaps 'There is no whitespace either before or after # in comment' can be divided into two parts and perhaps `Is there a typo here? If not ignore this message` should be added. :)